src/hooks/useVerifyCourseUnitHashes.tsx

```diff
@@ -0,0 +1,69 @@
+import useSWR from 'swr';
+import api from '../api/backend';
+import { CourseInfo } from '../@types';
+
+/**
+ * Fetcher function to get course unit hashes and verify them.
+ * @param courseUnits Array of course units with expected hashes
+ * @returns Object with course unit IDs, their hashes, and validity
+ */
+const fetchAndVerifyCourseUnitHashes = async (courseUnits: CourseInfo[]) => {
+  if (!courseUnits || courseUnits.length === 0) {
+    console.log('No course units provided');
+    return { currentHashes: {}, hashValidationMap: {} };
+  }
+
+  const ids = courseUnits.map(course => course.id);
+  console.log('Fetching and verifying course unit hashes for IDs:', ids);
+
+  try {
+
+    const response = await api.getCourseUnitHashes(ids);
+    const currentHashes: Record<number, string> = response;
+
+    const hashValidationMap: Record<number, boolean> = {};
+
+
+    courseUnits.forEach(course => {
+      const backendHash = currentHashes[course.id];
+      const isValid = backendHash === course.hash;
+      hashValidationMap[course.id] = isValid;
+
+
+    });
+
+    return {
+      currentHashes,
+      hashValidationMap,
+    };
```

Comment:
What is your opinion about just returning a `Set` of ids of the course units that are not valid, instead of returning the `currentHashes` and the validation map? Wouldn't it make simpler to detect which course units need to be changed?

Reply:
yes that would  be more efficient, I did it that way to be more explicit but it makes sense so I will change it. Thanks for pointing that out. Ps what time interval between requests would be reasonable?

Reply:
I think 5 minutes or so is fine, but we can discuss it further in the next meeting with the rest of the team